ARGO-105 Fix response headers on update/delete

kaggis opened the pull request:
- Issues Found: 
 - That in some cases in update/delete responses (Avail.Profiles) the response header wasn't correctly formatted (Fixed)
 - That when an id was provided for update/delete profile it was forwarded to the query before checking that the string is a valid ObjectID string. 
 - Added proper xml responses when a profile is properly deleted or updated (Were not implemented)
 - Added checks when poem definitions are missing in a profile to properly display results with an empty "poem" string

kaggis commented:
please review
/cc @pkoro @skanct

pkoro commented:
+1 from my side @kaggis . Once it is also tested on devel node we should merge in order to deploy to production.

kaggis commented:
Adding Unit testing in availabilty profiles requests. 
 - Init testing enviroment (test database, seed values)
 - Test Create Profile

...other tests in the commits to follow

kaggis commented:
Added response message when user submits malformed json data in POST,PUT

Added comprehensive unit tests:
- Create Av.Profiles
- Read Av.Profile List
- Update Av.Profiles
- Delete Av.Profiles
- Tests for using wrong profile ids
 - in Update request
 - in Delete request
- Tests for using bad json input
 - in Create request
 - in Update request
- Tests for bad authorization
 - in Create request
 - in Update request
 - in Delete request

Added comprehensive comments and go doc strings:


please review 
/cc @skanct @pkoro @dpavlos

dpavlos commented on app/availabilityProfiles/availabilityProfiles_test.go:
```diff
@@ -0,0 +1,555 @@
+	resp_bad_json       string
+}
+
+// Setup the Test Enviroment
```
Couple of minor typos eg "enviroment" "Also here is are instantiated"

dpavlos commented on app/availabilityProfiles/availabilityProfiles_test.go:
```diff
@@ -0,0 +1,555 @@
+}
+
+// Testing creation of a profile  using POST request.
+// During Setup of the test enviroment the testdb is seeded with
```
Typo "enviroment"

dpavlos commented on app/availabilityProfiles/availabilityProfiles_test.go:
```diff
@@ -0,0 +1,555 @@
+}
+
+// Testing Reading of profile list using GET request.
+// During Setup of the test enviroment the testdb is seeded with
```
typo

pkoro commented on app/availabilityProfiles/availabilityProfiles_test.go:
```diff
@@ -0,0 +1,555 @@
+            [
+                "service4",
+                "service5",
+                "service6"
```
Indentation here is a bit off (3 spaces instead of 2).

kaggis replied:
Regarding identation: it belongs inside a string litteral in opened ` `

pkoro commented:
+1 from me. My only comment other than the minor ones on typos and indentation is that maybe we should add in the README a one line on how to generate docs.
